lib/httpclient/http_client.go

```diff
@@ -0,0 +1,88 @@
+package httpclient
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+)
+
+type Client struct {
+	apiKey       string
+	clientSecret string
+	host         string
+	http         http.Client
+}
+
+func NewClient(host string, apiKey string, clientSecret string) *Client {
+	return &Client{
+		host:         host,
+		apiKey:       apiKey,       // pragma: allowlist secret
+		clientSecret: clientSecret, // pragma: allowlist secret
+	}
+}
+
+func (c *Client) SendRequest(ctx context.Context, endpoint string, httpMethod string, bodyJSON any, headers map[string]string) (string, error) {
+	endpoint = fmt.Sprintf("%s/%s", c.host, endpoint)
+	req, err := http.NewRequestWithContext(
+		ctx,
+		httpMethod,
+		endpoint,
+		nil,
+	)
+	if err != nil {
+		return "", errors.Wrap(err, "new http request")
+	}
+
+	bodyBytes, err := json.Marshal(bodyJSON)
+	if err != nil {
+		return "", errors.Wrap(err, "marshaling JSON")
+	}
+
+	if bodyJSON != nil {
+		req.Body = io.NopCloser(bytes.NewReader(bodyBytes))
+	}
+
+	req.Header = getHeaders(headers)
+
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return "", errors.Wrap(err, "http Do")
+	}
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			log.Error(ctx, "Http: closing body failure", err)
+		}
+	}()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", errors.Wrap(err, "read response body", "body")
+	}
+
+	bodyString := string(body)
+	log.Info(ctx, "Http: response body", "body", bodyString)
```

Comment:
in general, don't log in "libraries"

Reply:
good catch, missed removing this one from when I was testing